test/widgets/dialog_checks.dart

```diff
@@ -12,15 +14,32 @@ Widget checkErrorDialog(WidgetTester tester, {
   required String expectedTitle,
   String? expectedMessage,
 }) {
-  final dialog = tester.widget<AlertDialog>(find.byType(AlertDialog));
-  tester.widget(find.descendant(matchRoot: true,
-    of: find.byWidget(dialog.title!), matching: find.text(expectedTitle)));
-  if (expectedMessage != null) {
-    tester.widget(find.descendant(matchRoot: true,
-      of: find.byWidget(dialog.content!), matching: find.text(expectedMessage)));
+  switch (defaultTargetPlatform) {
+    case TargetPlatform.android:
+    case TargetPlatform.fuchsia:
+    case TargetPlatform.linux:
+    case TargetPlatform.windows: {
+      final dialog = tester.widget<Dialog>(find.byType(Dialog));
+      tester.widget(find.widgetWithText(Dialog, expectedTitle));
+      if (expectedMessage != null) {
+        tester.widget(find.widgetWithText(Dialog, expectedMessage));
+      }
+      return tester.widget(
+        find.descendant(of: find.byWidget(dialog),
+          matching: find.widgetWithText(TextButton, 'OK')));
+    }
+    case TargetPlatform.iOS:
+    case TargetPlatform.macOS: {
```

Comment:
nits: no need for braces; and use a blank line to visually separate the two case blocks, since they're both several lines long

```suggestion
          matching: find.widgetWithText(TextButton, 'OK')));

    case TargetPlatform.iOS:
    case TargetPlatform.macOS:
```